ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/platform.py

```diff
@@ -168,13 +168,15 @@ def default_interfaces(self: SharedUtils) -> dict:
         # First look for a matching default interface set that matches our platform and type
         for default_interface in default_interfaces:
             for platform in default_interface.get("platforms", []):
-                if search(f"^{platform}$", device_platform) and self.type in default_interface.get("types", []):
+                if search(f"^{platform}$", device_platform) and (
+                    self.type in default_interface.get("types", []) or "default" in default_interface.get("types", [])
+                ):
                     return default_interface
 
         # If not found, then look for a default default_interface that matches our type
         for default_interface in default_interfaces:
             for platform in default_interface.get("platforms", []):
-                if search(f"^{platform}$", "default") and self.type in default_interface.get("types", []):
+                if search(f"^{platform}$", "default") and (self.type in default_interface.get("types", []) or "default" in default_interface.get("types", [])):
```

Comment:
This logic will not work for all cases. It is suspectible to reordering changes. Since first match will win, so if default is before the more specific match it will take the default. I think you need another for loop.

Another problem:
```
# , platform , type
1 , default  , default
2 , abc      , default
3 , default  , xyz
4 , abc      , xyz
```
The two middle ones will lead to ambiguity. Which one will it take, since they are equally imprecise. I am not sure which behavior we want here.

Reply:
Yeah that makes sense.  I had considered if we just wanted to document that the first matching rule wins, or that any entries with 'default' in either list should be at the end of the list?  Let me try and make this more deterministic.

Reply:
Latest commit reorders the default_interfaces like so:
```
        # #  platform  type
        # 1  abc       xyz
        # 2  abc       default
        # 3  default   xyz
        # 4  default   default
```